#! /usr/bin/env python3

PACKAGE = 'bitbots_ros_control'
import roslib

roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#       Name       Type      Level    Description     Default Min   Max

gen.add("read_position", bool_t, 1,
        "Enable reading the current position values of the servos.")
gen.add("read_velocity", bool_t, 1,
        "Enable reading the current velocity values of the servos.")
gen.add("read_effort", bool_t, 1,
        "Enable reading the current effort values of the servos.")
gen.add("read_pwm", bool_t, 1,
        "Enable reading the current PWM values of the servos.")
gen.add("read_volt_temp", bool_t, 1,
        "Enable reading the current input voltages and temperature values of the servos.")
gen.add("VT_update_rate", int_t, 1,
        "How often the voltage and temperature should be read, i.e. putting 10 would mean every 10th time the position is read, VT is also read.", min=0, max=1000)
gen.add("warn_temp", double_t, 1,
        "Temperture threshold which starts throwing warnings in the motors [deg C].", min=0, max=100)
gen.add("warn_volt", double_t, 1,
        "Input voltage threshold which starts throwing warnings in the motors [V].", min=0, max=20)
exit(gen.generate(PACKAGE, "bitbots_ros_control", "dynamixel_servo_hardware_interface_params"))
